### ISSUE TITLE
Error: Module not found: Error: Can't resolve 'canvas' in 'path/to/node_modules/pdfjs-dist/es5/build'

### ISSUE THREAD

**Issue by yisalk** [VadimDez/ng2-pdf-viewer#788]
Hi great library to work with, I am facing the following issue with the latest version of `ng2-pdf-viewer` which `7.0.1` while creating the build by using the command `npm run build:ssr`.  

`./node_modules/pdfjs-dist/es5/build/pdf.js:14174:21-38 - Error: Module not found: Error: Can't resolve 'canvas' in '/path/to/node_modules/pdfjs-dist/es5/build''`

It was working fine with the previous version `6.4.1`.

I am working with angular 11.

Any suggestion of even a place to look to achieve the desired results?
Regards,

**Comment by tayambamwanza** [comment]
Same Problem for me

**Comment by srikanta72** [comment]
I am also facing same issue how to solve this issue?

**Comment by predycsoft** [comment]
Have you found a solution for this issue?

**Comment by cwooi95** [comment]
Is there any solution/ suggestion on this issue? I'm facing the same problem when trying to execute npm run build:ssr

**Comment by cwooi95** [comment]
I have tried adding canvas into external dependencies, but it pops out another error.

I have also encounter the issue with Angular Universal, because the server environment inherently does not have the same capabilities as the browser, so it does not recognize document in server side. 

Here's a doc you may refer to for that:
https://github.com/angular/universal/blob/master/docs/gotchas.md

**Comment by geekrohan** [comment]
@predycsoft Please share the solution here, it would be really helpful for all of us like me who is looking for the solution.

**Comment by tleperou** [comment]
+1 in here

**Comment by coremicroservices** [comment]
still I'm getting same issue.
![image](https://user-images.githubusercontent.com/25594305/173219337-1b53e2ae-713e-4057-8081-c6784260996a.png)

**Comment by mohammadghaderi** [comment]
Same Problem for me

**Comment by aishwhensheds** [comment]
Same Problem for me

**Comment by doyouwannatea** [comment]
Same Problem for me

**Comment by tanujlight** [comment]
same problem for me

**Comment by ODAncona** [comment]
It's a pleasure to see I'm not the only one to struggle with angular universal and pdf-js.

I fixed the document error with:

```
npm uninstall ng2-pdf-viewer
npm install --save ng2-pdf-viewer
npm update
```
But I've still no idea how to fix canvas error

Error: Can't resolve 'canvas' in /node_modules/pdfjs-dist/build'

**Comment by takion-platform** [comment]
Any updates?

**Comment by rahulexpo** [comment]
So I have been working on an angular app recently. Initially the app had angular 11 and I updated it to angular 14 without much problem. After that I implemented angular universal in it. And then I came across an ocean of bugs and issues. For this particular issue what I did was, **uninstall the package** and continue on fixing other issues and when angular universal started working I **installed it again** and it was not causing any issue.
